resolver/resolver.cc

```diff
@@ -527,7 +527,20 @@ class ResolveConstantsWalk {
         }
 
         auto *send = ast::cast_tree<ast::Send>(asgn->rhs.get());
-        if (send != nullptr && send->fun == core::Names::typeAlias() && send->args.size() == 1) {
+        if (send != nullptr && send->fun == core::Names::typeAlias()) {
+            if (send->args.size() == 0) {
```

Comment:
It looks like this case is a problem too:
```
B = T.type_alias(String,Integer)
```
Should you check for `size() != 1` instead?

Reply:
I think that's a question of what the behavior and error messages should look like. With the way this PR is currently written, that case is handled mostly fine, albeit with slightly different semantics and error messages. In particular, regardless of the `size()` of the arg list, it'll use the first argument to `type_alias` as the actual type alias for the purposes of constant resolution, which means in the above case it'll treat it as though we wrote `B = T.type_alias(String)`. Later on—during inference—we'll raise an error because `T.type_alias(String, Integer)` doesn't match the expected arity. So running my branch on that program currently produces the non-`ENFORCE`-ey output
```
/Users/gdritter/stripe/sorbet/samp.rb:2: Too many arguments provided for method T.type_alias. Expected: 1, got: 2 https://srb.help/7004
     2 |B = T.type_alias(Integer, String)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    https://github.com/sorbet/sorbet/tree/master/rbi/sorbet/t.rbi#L58: type_alias defined here
    58 |  def self.type_alias(type);
```
However, I can imagine an argument that I should check for `size() != 1` and in that case clear out the arguments to `T.type_alias` before producing a synthetic `untyped`. In that case, we'd be treating `T.type_alias(T1, T2, ...)` as though the user wrote `T.type_alias(T.untyped)` and then we could emit a custom error message here in the resolver that's not just the usual mismatched-arity message. I can see arguments both ways! Do you have a feeling one way or another?

Reply:
On master, the example with multiple arguments was raising an exception, so if it's producing an error now that's much better. I also like that it's just giving an error about the arity of `T.type_alias` now. Thanks!